### ISSUE TITLE
Join extension attributes are not added to Order results (REST api)

### ISSUE THREAD

**Issue by Bbbrinks** [magento/magento2#8035]
### Preconditions

1. Magento 2.1.3 
2. PHP 7.0.8-0ubuntu0.16.04.3 

### Steps to reproduce

1. Define a Extension attribute for the OrderInterface with a join table:
```
    <extension_attributes for="Magento\Sales\Api\Data\OrderInterface">
        <attribute code="some_id" type="string">
            <join reference_table="SOMETABLE" join_on_field="quote_id" reference_field="quote_id">
                <field>some_id</field>
            </join>
        </attribute>
    </extension_attributes>
```
2. Make sure there's matching data in the SOMETABLE table and the sales_order table
3. Request orders via REST

### Expected result

1. I expected to see some_id in the resulting json response (this does work for Quote extension attributes).

### Actual result

1. No some_id in the extension attributes property of the json response

**Comment by Bbbrinks** [comment]
As far as I can see there's a missing call in /magento/module-sales/Model/OrderRepository.php:

`$this->extensionAttributesJoinProcessor->process($this->quoteCollection);`

The //magento/module-quote/Model/QuoteRepository.php does have this call.

**Comment by PaulBoss** [comment]
Is there any update on this?

**Comment by magento-engcom-team** [comment]
@sterfpaul, thank you for your report.
We've created internal ticket(s) MAGETWO-81288 to track progress on the issue.

**Comment by magento-engcom-team** [comment]
Hi @sterfpaul. Thank you for your report. 
The issue has been fixed in magento-engcom/magento2ce#1168 by @nmalevanec in 2.2-develop branch
Related commit(s):
 - [0e8311ded322d3e50c9186544b1cd9a63929c79c](https://github.com/magento/magento2/commit/0e8311ded322d3e50c9186544b1cd9a63929c79c)

The fix will be available with the upcoming 2.2.4 release.

**Comment by sidolov** [comment]
Hi @sterfpaul. Thank you for your report. 
The issue has been fixed in magento/magento2#16169 by @Scarraban in 2.1-develop branch
Related commit(s):
 - [032c82d98b80e509ab8a63f93b6e169f37ce113d](https://github.com/magento/magento2/commit/032c82d98b80e509ab8a63f93b6e169f37ce113d)
 - [27bdf61e7c94d7bda0586094ebeca31fa7a60cc7](https://github.com/magento/magento2/commit/27bdf61e7c94d7bda0586094ebeca31fa7a60cc7)

The fix will be available with the upcoming 2.1.16 release.

**Comment by Ian410** [comment]
This seems to have been removed in 2.3, or was never included in the first place.  I can't find any commit removing it, why was it?

**Comment by magento-engcom-team** [comment]
Hi @Bbbrinks. Thank you for your report. 
The issue has been fixed in magento/magento2#21797 by @swnsma in 2.3-develop branch
Related commit(s):
 - [961e05c23d074b00ebf7c411d6bb677d467244be](https://github.com/magento/magento2/commit/961e05c23d074b00ebf7c411d6bb677d467244be)
 - [4603abc5e36637758ffc48ec93c7a08cd51b45b7](https://github.com/magento/magento2/commit/4603abc5e36637758ffc48ec93c7a08cd51b45b7)
 - [a45fb74cac0e9efa8a5ce9c304a8b55f1ce6e393](https://github.com/magento/magento2/commit/a45fb74cac0e9efa8a5ce9c304a8b55f1ce6e393)

The fix will be available with the upcoming 2.3.2 release.
